lib/cib/cib_client.c

```diff
@@ -390,23 +390,66 @@ cib_client_erase(cib_t * cib, xmlNode ** output_data, int call_options)
 static int
 cib_client_init_transaction(cib_t *cib, int call_options)
 {
+    int rc = pcmk_rc_ok;
+
     op_common(cib);
-    return cib_internal_op(cib, PCMK__CIB_REQUEST_INIT_TRANSACT, NULL, NULL,
-                           NULL, NULL, call_options, NULL);
+
+    if (cib->transaction != NULL) {
+        // A client can have at most one transaction at a time
+        rc = pcmk_rc_already;
+    }
+
+    cib->transaction = create_xml_node(NULL, T_CIB_TRANSACTION);
```

Comment:
I assume you want to skip this if rc != pcmk_rc_ok

Reply:
Thanks